distribution.md

```diff
@@ -155,3 +155,73 @@ Where:
     * contain an array of deprecated APIs:
       * should only contain `0.x` or major versions
       * should only contain APIs that correspond to a spec release
+
+### Builder
+
+The following specifies the artifact format for a buildpacks builder.
+
+A Builder is an OCI Image that provides a distributable build environment.
+
+A platform supporting the builder SHOULD allow users to configure the build environment with a provided builder.
+
+#### General Requirements
+
+The Builder image MUST contain an implementation of the [lifecycle](#lifecycle), and [build-time](#build-image) environment and MAY contain [buildpacks](#buildpackage). Platforms SHOULD use builders to ease the build process. 
+
+#### Environment Variables
+
+A Builder MUST be an extension of a Build Image, and MUST retain all the specified environment variables set on the original build image, as specified in the Build Image specifications.
+
+#### Labels
+
+A Builder MUST be an extension of a Build Image, and MUST retain all the specified Labels set on the original build image, as specified in the Build Image specifications.
+
+A Builder image MUST contain an implementation of the [lifecycle](#lifecycle), and MUST retain all the specified Labels set on the original Lifecycle image, as specified in the lifecycle distribution specifications.
+
+A Builder image MAY contain [buildpacks](#buildpackage), and MAY retain all the specified Labels set on the original buildpackage, as specified in the buildpackage specifications.
+
+The following labels MUST be set in the builder environment (through the image config's `Labels` field):
+
+| Label             | Description
+| --------          | --------
+| `io.buildpacks.builder.metadata`     | A JSON object representing Builder metadata
+| `io.buildpacks.buildpack.order`      | A JSON object representing the order of the buildpacks stored on the builder
+
+`io.buildpacks.builder.metadata` (JSON)
+
+```json
+{
+  "description": "<description>",
+  "buildpacks": [
+    {
+      "id": "<buildpack ID>",
+      "version": "<buildpack version>",
+      "homepage": "<buildpack homepage>",
+    }
+  ],
+  "createdBy": {
+    "name": "<tool name>",
+    "version": "<tool version>",
+  }
+}
+```
+
+The `createdBy` metadata is meant to contain the name and version of the tool that created the builder.
+
+`io.buildpacks.buildpack.order` (JSON)
+
+```json
+[
+  {
+    "group":
+      [
+        {
+          "id": "<buildpack ID>",
+          "version": "<buildpack version>",
+          "optional": false
+        }
+      ]
+  }
+]
+
+```
```

Comment:
Based on the previous unreleased builder spec, we did have a "Filesystem" section. Looking at it in depth I think most things are covered by _inheriting_ layers from the other components (ie. lifecycle binaries from lifecycle image, buildpack layers from buildpackages, etc).

The one file that I want to discuss on whether it is necessary is `order.toml`. Builders currently have that file on the filesystem and should match the `io.buildpacks.buildpack.order` label. In many cases, platforms override or create a new version of this file before providing it to the lifecycle. Do we need to keep a filesystem format of the order?

In a theoretical world where it didn't exist, this is how existing known platforms could handle it.

 * pack - can easily add that file as a layer (we do this already when buildpacks are added).
 * kpack - already [creates an order.toml dynamically](https://github.com/pivotal/kpack/blob/ffd8cd1caf1e852df4e714d1cb8b9f4e7dd89e1b/pkg/cnb/builder_builder.go#L250-L287).
 * tekton - as part of "prepare" could inspect the image config and generate the file.
 
 Personally, I'm not seeing the value or requiring it in both places.

Reply:
@sclevine / @jkutner / @natalieparellano / @jabrown85: Do you all have an option here?

Reply:
`pack` would do the same as Tekton and create an `order.toml` based on the image label.

Reply:
I think we should leave `order.toml` required. It doesn't seem great that a simple platform like tekton would have to read the label instead of being able to rely on it being in the builder itself. It means platforms MUST be able to read labels prior to executing builds on them. If the platform allows customers to bring their own builder, it is much easier to assume the the builder knows how to resolve the order itself and the platform would have less work to do.

it also makes testing a builder harder and more magic IMO. Today you can do something like this to test out the builder image without having to use `pack` or anything else special.

```
docker run -it -v $(pwd):/workspace --entrypoint /cnb/lifecycle/creator heroku/buildpacks:20 myimg
```

That wouldn't be possible if the file is removed.

Reply:
I concede to your argument @jabrown85. I do feel it's not optimal but can see your points.

What do we think about the format? As you define it @jabrown85, it makes it seem like the format should be part of the distribution spec and not the `platform` spec. Today, it defers out to a platform API defined [format][format] when it's embedded into a distributed package. I don't see how that would work in your scenarios. I would argue that the format should be defined by the distribution spec. The platform could still create a new one if need be but it should do so while considering the distribution version.

[format]: https://github.com/buildpacks/spec/blob/main/platform.md#ordertoml-toml

Reply:
I'll stand down from my contentious point above and defer it to a separate RFC if need be. 

@jimil749 do you think we can add the notion of `order.toml`, its location `/cnb/order.toml`, and reference to its [format][format] in this spec?

[format]: https://github.com/buildpacks/spec/blob/main/platform.md#ordertoml-toml

Reply:
@jromero, I will get to it asap

Reply:
@jromero sorry for the delay, but I've added a `Filesystem` section wherein I've mentioned that `order.toml` MUST be present in the builder fs and referenced the platform specification for the format.